src/configuration/utils.ts

```diff
@@ -0,0 +1,15 @@
+import { PermissionEnum } from "@saleor/types/globalTypes";
+import { IntlShape } from "react-intl";
+
+import { createConfigurationMenu } from ".";
+
+export const getConfigMenuItemsPermissions = (
+  intl: IntlShape
+): PermissionEnum[] =>
+  createConfigurationMenu(intl)
+    .reduce(
+      (prev, next) =>
+        prev.concat(next.menuItems.map(menuItem => menuItem.permissions)),
+      []
+    )
```

Comment:
If you're using `reduce` and the 2 conditions are met:
1. Output is an array
2. The length of the input and output array is the same

then it means it can be replaced with `.map`:
```suggestion
    .map(menu => menu.menuItems.map(menuItem => menuItem.permissions))
```

Reply:
@jwm0 I think I initially wrote something like you suggested byt the length of the output array is different compared to input

Reply:
Ok, in this case the final `flat` (or reduce you used before) is not needed. I'd also suggest to use the spread operator as I believe it makes it more readable and inline with the rest of the codebase:
```js
.reduce(
      (prev, { menuItems }) =>
        [...prev, ...menuItems.map(({ permissions })=> permissions)],
      []
    )
.reduce() /// <-- remove this part!!!
```